<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>502</width>
    <height>683</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>10</y>
      <width>434</width>
      <height>695</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="generalTab">
     <attribute name="title">
      <string>General Settings</string>
     </attribute>
     <widget class="QLabel" name="cameraLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>62</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Camera:</string>
      </property>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>110</y>
        <width>451</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="Line" name="line_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>320</y>
        <width>451</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="fullScreenLabel">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>20</y>
        <width>161</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Press F7 for Full Screen</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>149</width>
        <height>85</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="EngineLabel">
         <property name="text">
          <string>Engine:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="morphingEnabler">
         <property name="text">
          <string>Enable Morphing</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="rayTracerEnabler">
         <property name="text">
          <string>Enable Raytracing</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="particleSystemEnabler">
         <property name="text">
          <string>Light up the Candle</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>220</y>
        <width>231</width>
        <height>24</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QLabel" name="speedLabel">
         <property name="text">
          <string>Speed:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="speedHorizontalSlider">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>190</y>
        <width>181</width>
        <height>81</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QLabel" name="maxSpeedLabel">
         <property name="text">
          <string>Max Speed:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="maxSpeedTextInput"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>190</y>
        <width>221</width>
        <height>81</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <widget class="QLabel" name="maxAccelerationLabel">
         <property name="text">
          <string>Max Acceleration:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="maxAccelerationTextInput"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>250</y>
        <width>221</width>
        <height>81</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QLabel" name="frictionMagnitudeLabel">
         <property name="text">
          <string>Friction Magnitude:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="frictionMagnitudeTextInput"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>213</width>
        <height>26</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_9">
       <item>
        <widget class="QLabel" name="currentViewLabel">
         <property name="text">
          <string>Current View</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="currentViewComboBox">
         <item>
          <property name="text">
           <string>Perspective</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
     <zorder>layoutWidget</zorder>
     <zorder>layoutWidget</zorder>
     <zorder>layoutWidget</zorder>
     <zorder>layoutWidget</zorder>
     <zorder>layoutWidget</zorder>
     <zorder>layoutWidget</zorder>
     <zorder>cameraLabel</zorder>
     <zorder>line</zorder>
     <zorder>line_2</zorder>
     <zorder>fullScreenLabel</zorder>
    </widget>
    <widget class="QWidget" name="particleTab">
     <attribute name="title">
      <string>Particles</string>
     </attribute>
     <widget class="QCheckBox" name="freezeParticlesCheckBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>60</y>
        <width>131</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Freeze Particles</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>291</width>
        <height>31</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="numberOfParticlesLabel">
         <property name="text">
          <string>Number of Particles</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="numberOfParticlesSpinBox">
         <property name="maximum">
          <number>1000000000</number>
         </property>
         <property name="singleStep">
          <number>1</number>
         </property>
         <property name="value">
          <number>1000</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>110</y>
        <width>301</width>
        <height>26</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="flameEquationLabel">
         <property name="text">
          <string>Flame Equation</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="particleFieldFunctionComboBox">
         <item>
          <property name="text">
           <string>Flame</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Tornado</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Custom</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QGroupBox" name="flameParamsGroup">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>160</y>
        <width>300</width>
        <height>190</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>190</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>190</height>
       </size>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QPushButton" name="flameShowButton">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>150</y>
         <width>114</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Show</string>
       </property>
      </widget>
      <widget class="QLabel" name="attractorPositionLabel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>121</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Attractor Position:</string>
       </property>
      </widget>
      <widget class="QPushButton" name="flameDefaultButton">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>150</y>
         <width>114</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Default</string>
       </property>
      </widget>
      <widget class="QLabel" name="xFieldLabel">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>0</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>x</string>
       </property>
      </widget>
      <widget class="QLabel" name="yFieldLabel">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>0</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>y</string>
       </property>
      </widget>
      <widget class="QPlainTextEdit" name="xFieldTextInput">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>20</y>
         <width>31</width>
         <height>21</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>31</width>
         <height>21</height>
        </size>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
      </widget>
      <widget class="QLabel" name="zFieldLabel">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>0</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>z</string>
       </property>
      </widget>
      <widget class="QPlainTextEdit" name="zFieldTextInput">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>20</y>
         <width>31</width>
         <height>21</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>31</width>
         <height>21</height>
        </size>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
      </widget>
      <widget class="QPlainTextEdit" name="yFieldTextInput">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>20</y>
         <width>31</width>
         <height>21</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>31</width>
         <height>21</height>
        </size>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
      </widget>
      <widget class="QWidget" name="">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>50</y>
         <width>237</width>
         <height>92</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="scaleLabel">
          <property name="text">
           <string>Scale:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSlider" name="scaleSlider">
          <property name="minimumSize">
           <size>
            <width>191</width>
            <height>22</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>191</width>
            <height>22</height>
           </size>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="powerLabel">
          <property name="text">
           <string>Power:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSlider" name="powerSlider">
          <property name="minimumSize">
           <size>
            <width>191</width>
            <height>22</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>191</width>
            <height>22</height>
           </size>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="rangeLabel">
          <property name="text">
           <string>Range:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSlider" name="rangeSlider">
          <property name="minimumSize">
           <size>
            <width>191</width>
            <height>22</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>191</width>
            <height>22</height>
           </size>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <zorder>xFieldLabel</zorder>
      <zorder>attractorPositionLabel</zorder>
      <zorder>flameDefaultButton</zorder>
      <zorder>yFieldLabel</zorder>
      <zorder>xFieldTextInput</zorder>
      <zorder>zFieldLabel</zorder>
      <zorder>zFieldTextInput</zorder>
      <zorder>flameShowButton</zorder>
      <zorder>yFieldTextInput</zorder>
      <zorder></zorder>
      <zorder>flameShowButton</zorder>
      <zorder>attractorPositionLabel</zorder>
      <zorder>flameDefaultButton</zorder>
      <zorder>xFieldLabel</zorder>
      <zorder>yFieldLabel</zorder>
      <zorder>xFieldTextInput</zorder>
      <zorder>zFieldLabel</zorder>
      <zorder>zFieldTextInput</zorder>
      <zorder>yFieldTextInput</zorder>
      <zorder></zorder>
     </widget>
     <widget class="QGroupBox" name="userDefinedBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>160</y>
        <width>300</width>
        <height>190</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>190</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>190</height>
       </size>
      </property>
      <widget class="QPushButton" name="userDefinedShowButton">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>110</y>
         <width>114</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Show</string>
       </property>
      </widget>
      <widget class="QPushButton" name="userDefinedDefaultButton">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>110</y>
         <width>114</width>
         <height>32</height>
        </rect>
       </property>
       <property name="text">
        <string>Default</string>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidget_3">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>279</width>
         <height>85</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_3">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::FieldsStayAtSizeHint</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="VxLabel_2">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="maximumSize">
           <size>
            <width>71</width>
            <height>16</height>
           </size>
          </property>
          <property name="text">
           <string>Vx(x, y, z):</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPlainTextEdit" name="vxTextInput_2">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>201</width>
            <height>21</height>
           </size>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="tabChangesFocus">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="VyLabel_2">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="maximumSize">
           <size>
            <width>71</width>
            <height>16</height>
           </size>
          </property>
          <property name="text">
           <string>Vy(x, y, z):</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPlainTextEdit" name="vyTextInput_2">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>201</width>
            <height>21</height>
           </size>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="tabChangesFocus">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="VzLabel_2">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="maximumSize">
           <size>
            <width>71</width>
            <height>16</height>
           </size>
          </property>
          <property name="text">
           <string>Vz(x, y, z):</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPlainTextEdit" name="vzTextInput_2">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>201</width>
            <height>21</height>
           </size>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="tabChangesFocus">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <zorder>freezeParticlesCheckBox</zorder>
     <zorder>layoutWidget</zorder>
     <zorder>layoutWidget</zorder>
     <zorder>scaleLabel</zorder>
     <zorder>powerLabel</zorder>
     <zorder>rangeSlider</zorder>
     <zorder>rangeLabel</zorder>
     <zorder>powerSlider</zorder>
     <zorder>scaleSlider</zorder>
     <zorder>updateVectorFieldButton</zorder>
     <zorder>flameParamsGroup</zorder>
     <zorder>pushButton_3</zorder>
     <zorder>userDefinedBox</zorder>
    </widget>
    <widget class="QWidget" name="morphingTab">
     <attribute name="title">
      <string>Morphing</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>30</y>
        <width>331</width>
        <height>24</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_10">
       <item>
        <widget class="QLabel" name="morphPercentageLabel">
         <property name="text">
          <string>Morph Percentage</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="morphPercentageSlider">
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>70</y>
        <width>111</width>
        <height>58</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QRadioButton" name="WineBottleRadioButton">
         <property name="text">
          <string>Wine Bottle</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="WhiskyBottleRadioButton">
         <property name="text">
          <string>Whisky Bottle</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="raytraceTab">
     <attribute name="title">
      <string>Raytrace</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="sceneTab">
     <attribute name="title">
      <string>Scene</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>259</width>
        <height>80</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="activeObjectLabel">
             <property name="text">
              <string>Active Object:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="currentActiveObjectLabel">
             <property name="text">
              <string>None</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QPushButton" name="addObjectButton">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>114</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Add Object...</string>
      </property>
     </widget>
    </widget>
   </widget>
   <zorder>tabWidget</zorder>
   <zorder>pushButton_2</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>502</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
